sdk/servicebus/Microsoft.Azure.ServiceBus/src/ServiceBusConnectionStringBuilder.cs

```diff
@@ -260,6 +302,11 @@ public string GetNamespaceConnectionString()
                 connectionStringBuilder.Append($"{OperationTimeoutConfigName}{KeyValueSeparator}{this.OperationTimeout}{KeyValuePairDelimiter}");
             }
 
+            if (this.Authentication != null)
```

Comment:
This pattern of building negates a lot of the advantage of using a `StringBuilder`.   Unless there's a compiler optimization that I'm not aware of, you're essentially allocating another string builder to do the format for the bling string and feeding it's output into your string builder.   

You may want to consider just doing `AppendFormat` or a series of `Append` directly to your builder.